python-checks/src/main/java/org/sonar/python/checks/UndeclaredNameUsageCheck.java

```diff
@@ -46,22 +46,18 @@
 
 @Rule(key = "S3827")
 public class UndeclaredNameUsageCheck extends PythonSubscriptionCheck {
-  private static boolean hasWildcardImport = false;
-  private static boolean callGlobalsOrLocals = false;
   private SubscriptionContext subscriptionContext;
 
   @Override
   public void initialize(Context context) {
     context.registerSyntaxNodeConsumer(Tree.Kind.FILE_INPUT, ctx -> {
       subscriptionContext = ctx;
       FileInput fileInput = (FileInput) ctx.syntaxNode();
-      ExceptionVisitor exceptionVisitor = new ExceptionVisitor();
-      fileInput.accept(exceptionVisitor);
-      hasWildcardImport = exceptionVisitor.hasWildcardImport;
-      callGlobalsOrLocals = exceptionVisitor.callGlobalsOrLocals;
       UnresolvedSymbolsVisitor unresolvedSymbolsVisitor = new UnresolvedSymbolsVisitor();
       fileInput.accept(unresolvedSymbolsVisitor);
-      addNameIssues(unresolvedSymbolsVisitor.nameIssues);
+      if(!unresolvedSymbolsVisitor.callGlobalsOrLocals && !unresolvedSymbolsVisitor.hasWildcardImport) {
+        addNameIssues(unresolvedSymbolsVisitor.nameIssues);
```

Comment:
I think you can pass `ctx` as an argument to `addNameIssues` and drop the field named `subscriptionContext`.